### DISCUSSION TITLE
How to delete a user from supabase using Next.js 14?

### DISCUSSION THREAD

**Discussion by 43alsosc** [supabase/supabase#23144] [Questions]
So i am new to web development. So i want to be able to make a customer delete their own account from my website, which is stored under supabase's auth. I have created a script and a button to delete their account, but i do not know how to make them admin to allow them to delete their account.

Any help is much appreciated!
If you need any more context just ask :D

```
import { createClient } from "@/utils/supabase/server";

export async function DeleteUser() {
  const deleteAccount = async () => {
    "use server";
    const supabase = createClient();

    const { data, error } = await supabase.auth.getUser(); // Fetch user that is signed in

    if (error) {
      console.log("Error fetching user", error);
    }

    const userId = data?.user?.id; // Null check

    if (userId) {
      const { data, error } = await supabase.auth.admin.deleteUser(userId); // Delete the user
      if (error) {
        console.log("Error deleting user", error); // Log Error
      } else {
        await supabase.auth.signOut(); // Sign Out
        console.log("User deleted");
        revalidatePath(`/`); 
        redirect("/login");
      }
    }
  };
  return (
    <form action={deleteAccount}>
      <Button className="">Delete User</Button>
    </form>
  );
}
```

**Comment by pruthvi2103** [comment]
its always better to 'soft delete' things such as this. the auth.admin.deleteUser function requires a service key which is really unsafe to expose

**Reply by 43alsosc** [reply]
How would i do this? I use email, google and github to host accounts. I dont quite know what 'soft delete' is, i think i understand it being that i still have it, but it's set to 'deleted_at' and the user cannot sign in with it?

is using RLS policies the way to go for the user to not sign in if account is set to 'deleted' and store the deleted mail for the accounts in a table with ENUM (deleted, not_deleted, null(never_deleted))?
